### PR TITLE
fix(tests): Fix dnd backend test flakes

### PR DESCRIPTION
**Pull request by shruthilayaj**
This PR fixes 3 major flakes:

Fixes SENTRY-TESTS-3J5: Just sort the project id order

Fixes SENTRY-TESTS-3HQ: Flakes because we calculate the retention
in the test once and the value returned in the response is calculated
a little while after. We don't need to test for seconds granularity
so replacing seconds to 0.

Fixes SENTRY-TESTS-3J0: Successively calling before_now results in some flakes
particularly in tests that are calculating aggregates
on transaction.duration. Introduced a load_data method
that takes a datetime object `timestamp` and a timedelta `duration`
calculates the offset based on timestamp to get start_timestamp.

### CODE REVIEW

File: src/sentry/utils/samples.py [hunk 1]
```diff
@@ -180,8 +181,7 @@ def load_data(
 
     # Generate a timestamp in the present.
     if timestamp is None:
-        timestamp = datetime.utcnow() - timedelta(minutes=1)
-        timestamp = timestamp - timedelta(microseconds=timestamp.microsecond % 1000)
+        timestamp = before_now(minutes=1)
```
Comment (shruthilayaj) [inline src/sentry/utils/samples.py:184]: before_now does the same thing

File: tests/sentry/api/endpoints/test_organization_dashboard_details.py [hunk 2]
```diff
@@ -1364,7 +1366,7 @@ def test_update_dashboard_with_filters(self):
 
         response = self.do_request("put", self.url(self.dashboard.id), data=data)
         assert response.status_code == 200, response.data
-        assert response.data["projects"] == [project1.id, project2.id]
+        assert sorted(response.data["projects"]) == [project1.id, project2.id]
```
Comment (shruthilayaj) [inline tests/sentry/api/endpoints/test_organization_dashboard_details.py:1369]: SENTRY-TESTS-3J5

File: tests/sentry/api/endpoints/test_organization_dashboard_details.py [hunk 3]
```diff
@@ -201,7 +201,9 @@ def test_response_truncates_with_retention(self):
             response = self.do_request("get", self.url(dashboard.id))
 
         assert response.data["expired"]
-        assert iso_format(response.data["start"]) == expected_adjusted_retention_start
+        assert iso_format(response.data["start"].replace(second=0)) == iso_format(
```
Comment (shruthilayaj) [inline tests/sentry/api/endpoints/test_organization_dashboard_details.py:204]: SENTRY-TESTS-3HQ

File: tests/snuba/api/endpoints/test_organization_events.py [hunk 4]
```diff
@@ -57,6 +58,19 @@ def do_request(self, query, features=None):
         with self.feature(features):
             return self.client_get(self.reverse_url(), query, format="json")
 
+    def load_data(self, platform="transaction", timestamp=None, duration=None, **kwargs):
```
Comment (shruthilayaj) [inline tests/snuba/api/endpoints/test_organization_events.py:61]: SENTRY-TESTS-3J0

**Review by edwardgou-sentry — APPROVED** [review]

**Pull request merged**